### ISSUE TITLE
Scheduled jobs - `manifest.yaml` - when specifying `environments` map, top-level `on.schedule` setting is nulled out

### ISSUE THREAD

**Issue by camilosantana** [aws/copilot-cli#2043]
## Summary

**Scheduled jobs** `manifest.yaml` - when specifying `environments` map, top-level `on.schedule` setting is nulled out.

This isn't expected behaviour. [Documentation states](https://aws.github.io/copilot-cli/docs/manifest/scheduled-job/#environments) `environments` mapping is supported for jobs
as with any other service manifest.

## Problem

the following Scheduled Job fails when i add a matching environment in the environments map. manifest below.

```yaml
name: scheduled-job
type: Scheduled Job

on:
  schedule: "@hourly"

image:
  build: ./Dockerfile

environments:
  staging:
    variables:
      arbitrary-variable: arbitrary-value
```

```shell
copilot deploy \
  --app application \
  --env staging \
  --name scheduled-job

... builds container
... pushes layers

✘ Failed to deploy job.

✘ execute job deploy: deploy job: convert schedule for job scheduled-job: missing required field "schedule"
    in manifest for job scheduled-job
```

if i remove the environments map, it works fine.

## Workaround

If you specify `on.schedule` within the `environments` mapping, the job deploys without incident.

Example:

```yaml
name: scheduled-job
type: Scheduled Job
image:
  build: ./Dockerfile
environments:
  staging:
    on:
      schedule: "@hourly"
    variables:
      arbitrary-variable: arbitrary-value
```

the above deploys 👍

**Comment by camilosantana** [comment]
this is a verbose duplicate of #1736

**Comment by efekarakus** [comment]
Oh no! okay thanks for bringing this back into our attention again :bow: 

We sent a fix for it in #2044 and it should go out next release!

**Comment by efekarakus** [comment]
This is now released in v1.4.0! https://github.com/aws/copilot-cli/releases/tag/v1.4.0
